modules/appeals_api/spec/models/notice_of_disagreement_spec.rb

```diff
@@ -178,7 +180,50 @@
 
       notice_of_disagreement.update_status!(status: 'pending')
 
-      expect(handler).to have_received(:handle!)
+      expect(handler).to have_received(:handle!).exactly(1).times
+    end
+
+    it 'sends an email when emailAddressText is present' do
+      handler = instance_double(AppealsApi::Events::Handler)
+      allow(AppealsApi::Events::Handler).to receive(:new).and_return(handler)
+      allow(handler).to receive(:handle!)
+
+      notice_of_disagreement.update_status!(status: 'submitted')
+
+      expect(handler).to have_received(:handle!).exactly(2).times
+    end
+
+    it 'successfully gets the ICN when email isn\'t present' do
+      notice_of_disagreement = described_class.create!(
+        auth_headers: auth_headers,
+        form_data: form_data.deep_merge({
+                                          'data' => {
+                                            'attributes' => {
+                                              'veteran' => {
+                                                'emailAddressText' => nil
+                                              }
+                                            }
+                                          }
+                                        })
+      )
+
+      params = { event_type: :nod_received, opts: {
+        email_identifier: { id_value: '1013062086V794840', id_type: 'ICN' },
+        first_name: notice_of_disagreement.veteran_first_name,
+        date_submitted: notice_of_disagreement.date_signed,
```

Comment:
Whelp - this exposes that we've had a slightly incorrect test setup for a while across our HLR & SC notifiers. The `date_submitted` should be in ISO8601 format when sending to the event. Since the test only checks that we create a new Events::Handler, it doesn't expose that this date format isn't correct and may actually end up being a failure, since we use `Date.iso8601(date_submitted)` to parse the date out.

I think you can safely remove the `date_signed` method from the model, and then send in the iso8601 date here. I'll make a tech debt ticket to fix up the tests elsewhere.